code/__DEFINES/obj_flags.dm

```diff
@@ -11,6 +11,7 @@
 #define USES_TGUI				(1<<7)	//put on things that use tgui on ui_interact instead of custom/old UI.
 #define FROZEN					(1<<8)
 #define BLOCK_Z_FALL			(1<<9) // Should this object block z falling?
+#define AGE_RESTRICTED			(1<<10) ///Whether spessmen with an ID with an age below AGE_MINOR (20 by default) can buy this item in a vendor
```

Comment:
honestly i feel this would be better off as a component, that way you can have it things be blocked dynamically in the future when they have certain reagents in them, ie if you add vodka to a flask, the flask is then age restricted, but if there's water in it, it's fine, so on so forth.

Reply:
you cant because this is checked without initializing the object.